definitions/features/hammer.rb

```diff
@@ -109,12 +110,14 @@ def ask_password_and_check(custom_config)
 
   def config_error
     raise ForemanMaintain::HammerConfigurationError, 'Hammer configuration failed: '\
-                  "Is the admin password correct? (it was stored in #{custom_config_file})" \
+                  'Is the admin username and password correct? ' \
+                  "(it was stored in #{custom_config_file})" \
                   'Is the server down?'
   end
 
   def on_missing_password(custom_config)
-    if admin_password_missing?
+    username = custom_config[:foreman][:username]
+    if admin_password_missing? && (username == username_from_answers)
       msg = 'Admin password was not found in hammer configs. Looking into installer answers'
       logger.info(msg)
       custom_config[:foreman][:password] = password_from_answers
```

Comment:
We should test the username in here instead but Id suggest to test the initial_admin_user is the admin we account we need to use (explained above):
`if username_from_answers == feature(:foreman_server).admin_username`